install/helm/agones/templates/controller.yaml

```diff
@@ -13,6 +13,13 @@
 # limitations under the License.
 
 {{- $featureGates := include "agones.featureGates" . | fromYaml }}
+{{- $replicas := (int .Values.agones.controller.replicas) }}
+{{- if not $featureGates.SplitControllerAndExtensions }}
+{{- $replicas = 1 }}
+{{- end }}
+{{- if eq $replicas 0 }}
+{{- fail "replica field has an invalid value!" }}
```

Comment:
nit: full path of `agones.controller.replicas`, like you would set it on the command line.

Reply:
done